app/Http/Controllers/Assets/BulkAssetsController.php

```diff
@@ -53,6 +53,16 @@ public function edit(Request $request) : View | RedirectResponse
 
         $asset_ids = $request->input('ids');
         if ($request->input('bulk_actions') === 'checkout') {
+            $undeployable_assets = Asset::whereIn('id', $asset_ids)
+                ->whereHas('assetStatus', function ($query){
+                    $query->where('deployable', 0)
+                          ->where('archived', 0)
```

Comment:
I think we wouldn't want to allow an archived asset to be checked out?

I think there's a small conflict in a few of the ways we handle this, which we should probably address, since the behavior is different in a few spots. In the API Statuslabel `checkIfDeployable()` method, we allow pending or deployable.  (That is only used by the create/edit asset form when determining if we should present a checkout, via https://github.com/snipe/snipe-it/pull/16354)

In the query scopes on the Asset, `scopeRTD()`, `scopeUndeployable()`, we include/exclude pending. We should probably make those consistent across the board, but we'd need to see what else it might break before doing that.

Reply:
Sorry, the double negatives are killing me here lol

Reply:
I think deployable = 0 and archived = 0 is all we need here.

Reply:
If it is helpful, `availableForCheckout()` on the `Asset` model that would work in place of this query:
```php
$undeployable_assets = Asset::findMany($asset_ids)->filter(fn(Asset $asset) => !$asset->availableForCheckout());
```

Reply:
query has been changed, and it works 👍